embassy-stm32/src/lib.rs

```diff
@@ -1,7 +1,13 @@
 #![no_std]
 #![cfg_attr(
     feature = "nightly",
-    feature(type_alias_impl_trait, async_fn_in_trait, impl_trait_projections)
+    feature(
+        type_alias_impl_trait,
+        async_fn_in_trait,
+        impl_trait_projections,
+        const_mut_refs,
```

Comment:
We shouldn't add more nightly features. The plan is for the whole Embassy to work on stable once AFIT/TAIT is stable, which is hopefully near.

Reply:
So `new()` should just duplicate the code instead of using `init()`? Because there's no way to make `init()` `const` then.

Reply:
Yes... It's not too bad if we do zerofilling. `new` can use `mem::zeroed` and `init` can use `ptr::write_bytes(0)`

Reply:
I just noticed that `core::mem::zeroed()` it not `const`, so not really an option to use in `new()`...

Reply:
:( Then we can leave `new` as-is and then do zerofilling for `init` I guess

Reply:
Implemented in the last commit